### DISCUSSION TITLE
Aspira TV Front End

### DISCUSSION THREAD

**Discussion by simulot** [simulot/aspiratv#65] [Ideas]
Commençons en français. If english is required, let me know.

L'idée est de transformer AspiraTV en un service comme Emby, avec une installation simple, une configuration dans le navigateur, et configurer les séries à suivre facilement. 
- Configuration
  - Chemins, 
  - APIS KODI et autres...
  - langue préférée, sous-titres préférés,  
  - restrictions horaires... 
- Gestion des séries: 
  - Ajouter supprimer  une recherche,  
  - nettoyage des show récurrents, 
  - mise en pause du téléchargement ...

Techniquement, je veux continuer avec le langage GO et ses librairies standard.
AspraTV peut  servir les ressources  et expose une API
Le frontend pourrait être une SPA utilisant un framework récent
 
La distribution pourrait être:
- Un binaire auto suffisant et un procédure d'installation 
- Une image docker / docker compose

**Comment by LeoDupont** [comment]
Très bonne idée.
Des préférences pour le framework SPA ?
Selon moi, Vue a l'avantage d'être léger et simple, en plus d'être tendance ;)

**Reply by simulot** [reply]
Je voudrais  éviter de plonger dans le javascript pur et dur ...
L'application sera simpliste, et il n'y a pas d'enjeu, pas de deadline à tenir, ni de base installée... on peut explorer les options...

Idéalement, j'aimerai coder le frontend en Go.  Vugu https://www.vugu.org/doc  est prometteur.
- langage (trop) simple
- partage du code entre front et back end
- basé sur WASM qui n'est pas encore très sec

Vue est un bon compromis: 
- léger rapide
- bien établi
- typescript est un progrès par rapport à JS, mais ça reste du JS bien pimpé...

**Reply by LeoDupont** [reply]
Je ne connaissais pas Vugu, c'est intéressant.
Je suis partant pour essayer.

Et si on peut partager du code entre le front et le back c'est parfait.

**Comment by simulot** [comment]
## Réactivité du front end
Doit être bonne pour respecter l'utilisateur. 
- Eviter de scanner tous les dossiers pour répondre: trop long
- Doit être acceptable sur mobile et smart TV

## persistance des données
### Dans fichier JSON
| Les plus | Les moins |
|-|-|
|facile à mettre en oeuvre|ne grossi pas bien|
|éditable à la main|fragile|
|pas de dépendance| 

Bien pour la configuration statique:
- user / mdp
- chemin de la base de données


### Dans une base de donnée
| Les plus | Les moins |
|-|-|
|robuste |dépendance à installer|
|concurrence possible ||


Bien pour les données concernant les séries et épisodes
- épisodes déjà téléchargés
- métadonnée
-

**Comment by simulot** [comment]
J'ai testé go-app https://go-app.dev/start.

**Reply by simulot** [reply]
Moins de magie que VUGU, projet beaucoup plus ancien.
- Les composants sont écrits en GO directement.

Ca me convient mieux
